File: Leetcode/121 - Best Time to Buy and Sell Stock/solution.js
var maxProfit = function(prices) {
    let buyPrice = prices[0];
    let profit = 0;

    for (const price of prices) {
        if (price < buyPrice) {
            buyPrice = price;
        } else {
            profit = Math.max(profit, price - buyPrice);
        }
    }

    return profit;
};

File: Leetcode/121 - Best Time to Buy and Sell Stock/solution.cpp
int maxProfit(vector<int>& prices) {
    int buy_price = prices[0], profit = 0;
    for (auto& price: prices) {
        if (price < buy_price) {
            buy_price = price;
        } else {
            profit = max(profit, price - buy_price);
        }
    }
    return profit;
}